### PR TITLE
`py-nvidia-dali`: add v1.27.0

### PR DESCRIPTION
**Pull request by thomas-bouvier**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new version checksums for one of the architectures.

File: var/spack/repos/builtin/packages/py-nvidia-dali/package.py [hunk 1]
```diff
@@ -168,4 +194,6 @@ class PyNvidiaDali(PythonPackage):
     depends_on("python@3.6:3.11", when="@1.23:", type=("build", "run"))
     depends_on("python@3.6:3.10", when="@:1.22", type=("build", "run"))
     depends_on("py-astunparse@1.6.0:", type=("build", "run"))
-    depends_on("py-gast@0.2.1:0.4.0", type=("build", "run"))
+    depends_on("py-gast@0.2.1:0.4.0", when="@:1.26", type=("build", "run"))
+    depends_on("py-gast@0.3.3:", when="@1.27:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nvidia-dali/package.py]: I usually try to sort deps from newest to oldest like we do with versions

File: var/spack/repos/builtin/packages/py-nvidia-dali/package.py [hunk 2]
```diff
@@ -168,4 +194,6 @@ class PyNvidiaDali(PythonPackage):
...
+    depends_on("py-dm-tree", when="@1.27: ^python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nvidia-dali/package.py]:
```suggestion
    depends_on("py-dm-tree", when="@1.27:", type=("build", "run"))
```
Spack only supports Python 3.7+ anyway

File: var/spack/repos/builtin/packages/py-nvidia-dali/package.py [hunk 3]
```diff
@@ -151,13 +175,15 @@ class PyNvidiaDali(PythonPackage):
         "1.24.0-cuda120",
         "1.25.0-cuda120",
         "1.26.0-cuda120",
+        "1.27.0-cuda120",
     )
     cuda110_versions = (
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nvidia-dali/package.py:180]: Probably ought to reverse the order of these too just to be consistent.

**Review by adamjstewart — DISMISSED** [review]
Just a few very minor, very optional style things. If you're busy and just want me to merge this I'm also fine with that. Thanks for the update!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**